Mathlib/Algebra/Order/SuccPred/WithBot.lean

```diff
@@ -0,0 +1,30 @@
+/-
+Copyright (c) 2024 Yaël Dillies Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies Andrew Yang
+-/
+import Mathlib.Algebra.Order.Monoid.Unbundled.WithTop
+import Mathlib.Algebra.Order.SuccPred
+import Mathlib.Order.SuccPred.WithBot
+
+/-!
+# Algebraic properties of the successor function on `WithBot`
+-/
+
+namespace WithBot
+variable {α : Type*} [Preorder α] [OrderBot α] [AddMonoidWithOne α] [SuccAddOrder α]
+
+lemma succ_natCast (n : ℕ) : succ (n : WithBot α) = n + 1 := by
+  rw [← WithBot.coe_natCast, succ_coe, Order.succ_eq_add_one]
+
+@[simp] lemma succ_zero : succ (0 : WithBot α) = 1 := by simpa using succ_natCast 0
+
+@[simp]
+lemma succ_one : succ (1 : WithBot α) = 2 := by simpa [one_add_one_eq_two] using succ_natCast 1
+
+@[simp]
+lemma succ_ofNat (n : ℕ) [n.AtLeastTwo] :
+    succ (no_index OfNat.ofNat n : WithBot α) = OfNat.ofNat n + 1 := succ_natCast n
```

Comment:
```suggestion
    succ (no_index (OfNat.ofNat n) : WithBot α) = OfNat.ofNat n + 1 := succ_natCast n
```

Reply:
else this won't apply as you expect!

Reply:
Oh right! I thought this was precisely the case where it would work